cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -380,8 +380,8 @@ impl YaccParser<'_> {
                 i = self.parse_ws(j, false)?;
                 while i < self.src.len() && self.lookahead_is("%", i).is_none() {
                     let (j, n, span, _) = self.parse_token(i)?;
-                    self.ast.token_directives.insert(n.clone());
-                    if self.ast.tokens.insert(n) {
+                    if let (idx, true) = self.ast.tokens.insert_full(n) {
+                        self.ast.token_directives.insert(idx);
```

Comment:
I think this should actually go outside the if statement, we don't actually appear to insert "a" into tokens yet in the case of something like the following where "a" appears in `%left "a"` before `%token a`, however this code is kind of written in a way that the token could already be in `self.tokens`.  So I think it is better to put this call outside the `if`.

```
%left "a"
%token a
```

Reply:
should be fixed in [af08b7b](https://github.com/softdevteam/grmtools/pull/558/commits/af08b7bbba3ac3ffce9531d3f27020a81c022f63)